### PR TITLE
Drop Python 3.8 and run Python 3.9 tests/builds

### PR DESCRIPTION
**Pull request by shwina**

### CODE REVIEW

**Review by madsbk — APPROVED** [review]
LGTM, thanks @shwina

**Review by wence- — COMMENTED** [review]
sorry, I was too late...

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
     if: github.ref_type == 'branch' && github.event_name == 'push'
     needs: python-build
     secrets: inherit
-    uses: rapidsai/shared-action-workflows/.github/workflows/custom-job.yaml@branch-23.06
+    uses: rapidsai/shared-action-workflows/.github/workflows/custom-job.yaml@py-39
```
[the remark this answers is not in the document]
Comment (ajschmidt8) [reply .github/workflows/build.yaml:59]: short-lived.

the changes on https://github.com/rapidsai/shared-action-workflows/tree/py-39 will be merged into https://github.com/rapidsai/shared-action-workflows/tree/branch-23.06 once all of RAPIDS is successfully working with the `py-39` branch.

then we will open subsequent PRs to revert these branch changes.

[the remark this answers is not in the document]
Comment (wence-) [reply .github/workflows/build.yaml:59]: Thanks.

**Pull request merged**